vkconfig/active_layers_widget.cpp

```diff
@@ -228,8 +290,52 @@ bool ActiveLayersWidget::shouldClearOnClose()
     return cleanup_box->isChecked();
 }
 
-void ActiveLayersWidget::clearLayers()
-{
+void ActiveLayersWidget::clearDisabledLayers() {
+    while (!disabled_layers.isEmpty()) {
+        LayerManifest manifest = disabled_layers.first();
+        QListWidgetItem *item = disabled_layer_list->item(0);
+        QListWidget *explicit_widget = layer_lists[LayerType::Explicit];
+        QListWidget *implicit_widget = layer_lists[LayerType::Implicit];
+        bool found = false;
+        switch (manifest.type) {
+            case LayerType::Explicit:
+                for (int i = 0; i < explicit_widget->count(); ++i) {
+                    if (manifest < unset_layers[i]) {
+                        explicit_widget->insertItem(i, item->text());
+                        unset_layers.insert(i, manifest);
+                        found = true;
+                        break;
+                    }
+                }
+                if (!found) {
+                    explicit_widget->addItem(item->text());
+                    unset_layers.insert(explicit_widget->count(), manifest);
+                }
+                break;
+            case LayerType::Implicit:
+                for (int i = explicit_widget->count(); i < unset_layers.count(); ++i) {
```

Comment:
Nope. Since that list includes explicit then implicit layers, the start of the implicit laters should be after all explicit layers. That being said, this is probably a good place for a comment to that effect.